cin_validator/rules/cin2022_23/rule_8615.py

```diff
@@ -0,0 +1,153 @@
+from typing import Mapping
+
+import pandas as pd
+
+from cin_validator.rule_engine import rule_definition, CINTable, RuleContext
+from cin_validator.rule_engine import IssueLocator
+from cin_validator.test_engine import run_rule
+
+Section47 = CINTable.Section47
+S47ActualStartDate = Section47.S47ActualStartDate
+DateOfInitialCPC = Section47.DateOfInitialCPC
+LAchildID = Section47.LAchildID
+
+# define characteristics of rule
+@rule_definition(
+    code="8615",
+    module=CINTable.Section47,
+    message="Section 47 Enquiry Start Date must be present and cannot be later than the date of the initial Child Protection Conference",
+    affected_fields=[S47ActualStartDate, DateOfInitialCPC],
+)
+def validate(
+    data_container: Mapping[CINTable, pd.DataFrame], rule_context: RuleContext
+):
+    df = data_container[Section47]
+
+    """
+    Within a <Section47> group, if <DateOfInitialCPC> (N00110) is present then <S47ActualStartDate> (N00148) must be present and on or before the <DateOfInitialCPC> (N00110)
+    """
+
+    df.index.name = "ROW_ID"
+
+    df.query(
+        "(S47ActualStartDate > DateOfInitialCPC) or (DateOfInitialCPC.notna() and S47ActualStartDate.isna())",
+        inplace=True,
+    )
+
+    df_issues = df.reset_index()
+
+    link_id = tuple(
+        zip(
+            df_issues[LAchildID],
+            df_issues[S47ActualStartDate],
+            df_issues[DateOfInitialCPC],
+        )
+    )
+    df_issues["ERROR_ID"] = link_id
+    df_issues = df_issues.groupby("ERROR_ID")["ROW_ID"].apply(list).reset_index()
+
+    rule_context.push_type_1(
+        table=Section47,
+        columns=[S47ActualStartDate, DateOfInitialCPC],
+        row_df=df_issues,
+    )
+
+
+def test_validate():
+    IDS_are = [
+        "AAAAAAAA",
+        "BBBBBBBBB",
+        "CCCCCCCCCCC",
+        "DDDDDDDDD",
+        "EEEE",
+        "FFFFFFFFF",
+        "GGGGGGGGGG",
+        "HHHH",
+    ]
+    S47start = [
+        "01-01-2020",
+        "01-02-2020",
+        "01-03-2020",
+        "15-01-2020",
+        pd.NA,
+        "01-07-2020",
+        "15-01-2020",
+        pd.NA,
+    ]
+    ICPCstart = [
+        "01-01-2020",
+        "01-01-2020",  #  Fails as ICPC before S47.
```

Comment:
Added some comments here to show which rows failed and why to make review easier.